llvm/lib/Target/NVPTX/NVPTXIntrinsics.td

```diff
@@ -789,6 +789,17 @@ def PREFETCH_GLOBAL_L2_EVICT_LAST   : NVPTXInst<(outs), (ins Int64Regs:$addr),
 
 def PREFETCHU_L1 : PREFETCH_INTRS<"prefetchu.L1">;
 
+//Applypriority intrinsics
+class APPLYPRIORITY_L2_INTRS<string InstName> :
+          NVPTXInst<(outs), (ins Int64Regs:$addr, Int64Regs:$size),
+          InstName # " [$addr], $size;",
+          [(!cast<Intrinsic>("int_nvvm_" # !subst("::", "_", !subst(".", "_", InstName)))
+          i64:$addr, i64:$size)]>,
+          Requires<[hasPTX<74>, hasSM<80>]>;
+
+def APPLYPRIORITY_L2_EVICT_NORMAL        : APPLYPRIORITY_L2_INTRS<"applypriority.L2::evict_normal">;
+def APPLYPRIORITY_GLOBAL_L2_EVICT_NORMAL : APPLYPRIORITY_L2_INTRS<"applypriority.global.L2::evict_normal">;
```

Comment:
Instead of accepting complete instruction name, we could accept address space part only (I.e. `""` or `"global"`) and avoid having to replace dots/colons with underscores.

Reply:
Yes addressed. (used StrJoin)